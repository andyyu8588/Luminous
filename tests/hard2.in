// For compiler/VM testing purpose

function collatz(a) {
    print(a);
    if(a equals 1) {
        return;
    }
    else if(a % 2 equals 0) {
        collatz(a / 2);
    }
    else {
        collatz(3 * a + 1);
    }
}
print(collatz);
collatz(15);
collatz(50);
collatz(123);

a = 1;
b = 2;
c = 3;
d = 4;
e = 5;
f = 6;
g = 7;

function hard2Print(aa, bb, cc, dd, ee, ff, gg) {
    print(aa);
    print(bb);
    print(cc);
    print(dd + ee);
    print(ee - ff + gg);
    a = aa + b;
    c = c + dd;
    e = ff - gg;
}
print(hard2Print);
hard2Print(0, -1, 2, 5, -10, 20, 16);

print(a);
print(c);
print(e);
print(clock);
