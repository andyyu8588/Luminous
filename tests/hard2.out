collatz
15
46
23
70
35
106
53
160
80
40
20
10
5
16
8
4
2
1
50
25
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
123
370
185
556
278
139
418
209
628
314
157
472
236
118
59
178
89
268
134
67
202
101
304
152
76
38
19
58
29
88
44
22
11
34
17
52
26
13
40
20
10
5
16
8
4
2
1
hard2Print
0
-1
2
-5
-14
2
8
4
clock
